ui/helpers/utils/util.js

```diff
@@ -443,3 +443,109 @@ export const toHumanReadableTime = (t, milliseconds) => {
 export function clearClipboard() {
   window.navigator.clipboard.writeText('');
 }
+
+const solidityTypes = () => {
+  const types = [
+    'bool',
+    'address',
+    'string',
+    'bytes',
+    'int',
+    'uint',
+    'fixed',
+    'ufixed',
+  ];
+
+  const ints = Array.from(new Array(32)).map(
+    (_, index) => `int${(index + 1) * 8}`,
+  );
+  const uints = Array.from(new Array(32)).map(
+    (_, index) => `uint${(index + 1) * 8}`,
+  );
+  const bytes = Array.from(new Array(32)).map(
+    (_, index) => `bytes${index + 1}`,
+  );
+
+  /**
+   * fixed and ufixed
+   * This value type also can be declared keywords such as ufixedMxN and fixedMxN.
+   * The M represents the amount of bits that the type takes,
+   * with N representing the number of decimal points that are available.
+   *  M has to be divisible by 8, and a number from 8 to 256.
+   * N has to be a value between 0 and 80, also being inclusive.
+   */
+  const fixedM = Array.from(new Array(32)).map(
+    (_, index) => `fixed${(index + 1) * 8}`,
+  );
+  const ufixedM = Array.from(new Array(32)).map(
+    (_, index) => `ufixed${(index + 1) * 8}`,
+  );
+  const fixed = Array.from(new Array(80)).map((_, index) =>
+    fixedM.map((aFixedM) => `${aFixedM}x${index + 1}`),
+  );
+  const ufixed = Array.from(new Array(80)).map((_, index) =>
+    ufixedM.map((auFixedM) => `${auFixedM}x${index + 1}`),
+  );
+
+  return [
+    ...types,
+    ...ints,
+    ...uints,
+    ...bytes,
+    ...fixed.flat(),
+    ...ufixed.flat(),
+  ];
+};
+
+export const sanitizeMessage = (msg, baseType, types) => {
+  let baseTypeDefinitions;
+  if (types) {
+    baseTypeDefinitions = types[baseType];
+    if (!baseTypeDefinitions) {
+      return msg;
+    }
+  } else {
+    return msg;
+  }
+
+  const sanitizedMessage = {};
+  const msgKeys = Object.keys(msg);
+  msgKeys.forEach((msgKey) => {
+    const definedType = Object.values(baseTypeDefinitions).find(
+      (baseTypeDefinition) => baseTypeDefinition.name === msgKey,
+    );
+
+    if (!definedType) {
+      return;
+    }
+
+    // key has a type. check if the definedType is also a type
+    const nestedTypeDefinition = types[definedType.type.replace('[]', '')];
+
+    if (nestedTypeDefinition) {
+      if (definedType.type.indexOf('[]') > 0) {
+        // nested array
+        const definedArrayType = definedType.type.replace('[]', '');
+        sanitizedMessage[msgKey] = msg[msgKey].map((value) =>
+          sanitizeMessage(value, definedArrayType, types),
+        );
```

Comment:
Same thing here re: ensuring the type ends in `[]`
```suggestion
    const nestedTypeDefinition = types[definedType.type.replace(/\[\]$/, '')];

    if (nestedTypeDefinition) {
      if (definedType.type.endsWith('[]')) {
        // nested array
        const definedArrayType = definedType.type.replace(/\[\]$/, '')];
        sanitizedMessage[msgKey] = msg[msgKey].map((value) =>
          sanitizeMessage(value, definedArrayType, types),
        );
```

Or, to cache the nested type name:

```suggestion
    const nestedType = definedType.type.replace(/\[\]$/, '');
    const nestedTypeDefinition = types[nestedType];

    if (nestedTypeDefinition) {
      if (definedType.type.endsWith('[]')) {
        // nested array
        sanitizedMessage[msgKey] = msg[msgKey].map((value) =>
          sanitizeMessage(value, nestedType, types),
        );
```